### DISCUSSION TITLE
Rpc function not working

### DISCUSSION THREAD

**Discussion by ricardosikic** [supabase/supabase#12422] [Questions] [answered]
Hello, I am creating an rpc function to get the number of likes for each post for a user, so I have created a function that takes userId as an argument, which is the uuid of the user that is in the session. But when I call the function on the front it returns me.

`If a new function was created in the database with this name and parameters, try reloading the schema cache.`

The function:

```
create function get_number_of_posts_by_user(userId uuid)
returns integer
as $$
  SELECT 
  count(pl.id)
  FROM 
  auth.users au
  join posts p on p.user_id = au.id
  join post_likes pl on pl.post_id = p.id
  where au.id = userId
$$ language sql;

```

**Comment by GaryAustin1** [comment, marked as answer]
A couple of things.   You used userId without double quotes and so that is converted to userid in Postgres.  "userId" would keep the capital.  You really should not use capital letters in Postgres.  Then if you called the function from supabase-js with userId it would not be found.

You can't access auth.users unless you declare your function security definer.

It is better to use auth.uid() in your function if you want just the current user's id.  Passing in a uuid can be faked.

**Reply by ricardosikic** [reply]
thanks!
